src/MatrixTwitter.js

```diff
@@ -0,0 +1,987 @@
+    client.post("statuses/update",status,(error,tweet) => {
+      if(error){
+        log.error("Twitter","Failed to send tweet. %s",error);
+        return;
+      }
+      var id = sender.getId();
+      log.info("Twitter","Tweet sent from %s!",id);
+    });
+  }).catch(err =>{
+    log.error("Twitter","Failed to send tweet. %s",err);
+  });
+}
+
+/*
+  Timeline functions
+*/
+
+MatrixTwitter.prototype._process_timeline = function() {
+    if (this.timeline_queue.length < 1) {
+        return;
+    }
+
+
+    var tline = this.timeline_queue.shift();
+    var id = tline.entry.remote.getId().substr("@twitter_".length);
+
+    var req = {
+        user_id: id,
+        count: 3
+    };
+    var since = tline.entry.remote.get("twitter_since");
+    if (since != undefined) {
+        req.since_id = since;
+    }
+
+    this.app_twitter.get('statuses/user_timeline', req, (error, feed, response) => {
+        if(error){
+          log.error("Twitter","_process_timeline: GET /statuses/user_timeline returned: %s", error.msg);
+          return;
+        }
+        if (feed.length > 0) {
+            if(this.msg_queue_intervalID != null){
+              clearInterval(this.msg_queue_intervalID);
+              this.msg_queue_intervalID = null;
+            }
+            tline.entry.remote.set("twitter_since", feed[0].id_str);
+            var promises = [];
+            feed.reverse().forEach((item) => {
+                promises.push(this.process_tweet(tline.entry.matrix.roomId, item, TWEET_REPLY_MAX_DEPTH));
+            });
+            Promise.all(promises).then(() =>{
+              this._bridge.getRoomStore().upsertEntry(tline.entry);
+              this.msg_queue_intervalID = setInterval(() => {this._process_head_of_msg_queue();}, TWITTER_MSG_QUEUE_INTERVAL_MS);
+            });
+
+        }
+    });
+    this.timeline_queue.push(tline);
+}
+
+/**
+ * MatrixTwitter.prototype.add_timeline - Add a user's timeline to the timeline
+ * processor. Tweets will be automatically send to the given room.
+ *
+ * @param  {string} userid The Matrix user_id of the timeline owner. This would
+ *                         be a bridge user.
+ * @param  {external:Entry} entry The entry for a room in the RoomBridgeStore.
+ */
+MatrixTwitter.prototype.add_timeline = function(userid, entry) {
+    var obj = {
+        "user_id": userid,
+        "entry":entry
+    };
+    this.timeline_list.push(obj);
+    this.timeline_queue.push(obj);
+    log.info('Twitter',"Added Timeline: %s",userid);
+}
+
+
+/**
+ * MatrixTwitter.prototype.remove_timeline - Remove a timeline from being
+ * processed.
+ *
+ * @param  {type} userid The Matrix user_id of the timeline owner. This would
+ *                       be a bridge user.
+ */
+MatrixTwitter.prototype.remove_timeline = function(userid){
+  const tlfind = (tline) => { tline.user_id == userid };
+  this.timeline_list  = this.timeline_list.splice(this.timeline_list.findIndex(tlfind),1);
```

Comment:
What does `this.timeline_list` actually **do**? You add and remove from it, but seemingly never read from it?